static/js/components/SlackPreferences.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import FormGroup from "@material-ui/core/FormGroup";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Switch from "@material-ui/core/Switch";
+import IconButton from "@material-ui/core/IconButton";
+import TextField from "@material-ui/core/TextField";
+
+import HelpOutlineIcon from "@material-ui/icons/HelpOutline";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/core/styles";
+import Popover from "@material-ui/core/Popover";
+
+import * as profileActions from "../ducks/profile";
+
+const useStyles = makeStyles((theme) => ({
+  header: {
+    display: "flex",
+    alignItems: "center",
+    "& > h6": {
+      marginRight: "0.5rem",
+    },
+  },
+  typography: {
+    padding: theme.spacing(2),
+  },
+  textField: {
+    marginLeft: theme.spacing(1),
+    marginRight: theme.spacing(1),
+    "& p": {
+      color: "red",
+    },
+  },
+}));
+
+const SlackPreferences = () => {
+  const classes = useStyles();
+  const slack_preamble = useSelector((state) => state.sysInfo.slack_preamble);
+  const profile = useSelector((state) => state.profile.preferences);
+  const dispatch = useDispatch();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [slackurl, setSlackurl] = useState(profile.slack_integration?.url);
+  const [slackurlerror, setSlackurlerror] = useState(false);
+
+  const handleChange = (event) => {
+    setSlackurl(event.target.value);
+  };
+
+  const handleBlur = () => {
+    if (slackurl?.startsWith(slack_preamble)) {
+      setSlackurlerror(false);
+      const prefs = {
+        slack_integration: {
+          url: slackurl,
+        },
+      };
+      dispatch(profileActions.updateUserPreferences(prefs));
```

Comment:
How many times will the blur execute while typing, or while pasting?  Currently, we only have one other section of the profile that requires text entry, and that section uses a submit button.

Reply:
blur happens when focus in the textfield focus is lost. Without adding a submit button this was the way I decided to implement the error checking and the subsequent saving of the data. Open to other ideas.